src/CommunityToolkit.Maui/Converters/IsStringNotNullOrWhiteSpaceConverter.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using CommunityToolkit.Maui.Extensions;
+
+namespace CommunityToolkit.Maui.Converters;
+
+/// <summary>
+/// Converts the incoming value to a <see cref="bool"/> indicating whether or not the value is not null and not white space.
+/// </summary>
+public class IsStringNotNullOrWhiteSpaceConverter : ValueConverterExtension, ICommunityToolkitValueConverter
+{
+	/// <summary>
+	/// Converts the incoming string to a <see cref="bool"/> indicating whether or not the value is not null and not white space using string.IsNullOrWhiteSpace.
+	/// </summary>
+	/// <param name="value">The value to convert.</param>
+	/// <param name="targetType">The type of the binding target property. This is not implemented.</param>
+	/// <param name="parameter">Additional parameter for the converter to handle. This is not implemented.</param>
+	/// <param name="culture">The culture to use in the converter. This is not implemented.</param>
+	/// <returns>A <see cref="bool"/> indicating if the incoming value is not null and not white space.</returns>
+	public object Convert(object? value, Type? targetType, object? parameter, CultureInfo? culture)
+	{
+		if (value is null)
+		{
+			return false;
+		}
+
+		return value is string stringValue ?
+			!string.IsNullOrWhiteSpace(stringValue) :
+			throw new ArgumentException("Binding must be of type string");
+	}
```

Comment:
Hey Vlad! When I did the `IsStringNullOrEmptyConverter`, instead of using pattern matching, I cast `value` to a nullable string:

https://github.com/CommunityToolkit/Maui/blob/d3abd5e7bcd662f8f5d7164a500b2391712da580/src/CommunityToolkit.Maui/Converters/IsStringNotNullOrEmptyConverter.shared.cs#L20-L26

This throws an `InvalidCastException` if the user passes in anything other than a `string`, which I validate in the Unit Tests:
https://github.com/CommunityToolkit/Maui/blob/d3abd5e7bcd662f8f5d7164a500b2391712da580/src/CommunityToolkit.Maui.UnitTests/Converters/IsStringNullOrEmptyConverter_Tests.cs#L23-L32

I'm fine with either pattern, but let's pick one so that we're consistent. 

Let's either update `IsStringNullOrEmptyConverter` and `IsStringNotNullOrEmptyConverter` to throw `ArgumentException` in this PR, or we can use `InvalidCastException` for `IsString*WhiteSpaceConverter`👍